prefect_gcp/cloud_storage.py

```diff
@@ -665,3 +665,391 @@ async def write_path(self, path: str, content: bytes) -> str:
+                    gcs_bucket.download_object_to_file_object("my_folder/notes.txt", f)
+            ```
+        """
+        bucket = await self.get_bucket()
+
+        bucket_path = self._join_bucket_folder(from_path)
+        blob = bucket.blob(bucket_path)
+        self.logger.info(
+            f"Downloading blob from bucket {self.bucket!r} path {bucket_path!r}"
+            f"to file object."
+        )
+
+        await run_sync_in_worker_thread(
+            blob.download_to_file, file_obj=to_file_object, **download_kwargs
+        )
+        return to_file_object
+
+    @sync_compatible
+    async def download_folder_to_path(
+        self,
+        from_folder: str,
+        to_folder: Optional[Union[str, Path]] = None,
+        **download_kwargs: Dict[str, Any],
+    ) -> Path:
+        """
+        Downloads objects *within* a folder (excluding the folder itself)
+        from the object storage service to a folder.
+
+        Args:
+            from_folder: The path to the folder to download from; this gets prefixed
+                with the bucket_folder.
+            to_folder: The path to download the folder to. If not provided, will default
+                to the current directory.
+            **download_kwargs: Additional keyword arguments to pass to
+                `Blob.download_to_filename`.
+
+        Returns:
+            The path that the folder was downloaded to.
+
+        Examples:
+            Download my_folder to a folder named my_folder.
+            ```python
+            from prefect_gcp.cloud_storage import GcsBucket
+
+            gcs_bucket = GcsBucket.load("my-bucket")
+            gcs_bucket.download_folder_to_path("my_folder", "my_folder")
+            ```
+        """
+        if to_folder is None:
+            to_folder = ""
+        to_folder = Path(to_folder)
+
+        blobs = await self.list_blobs(folder=from_folder)
+        if len(blobs) == 0:
+            self.logger.warning(
+                f"No blobs were downloaded from "
+                f"bucket {self.bucket!r} path {from_folder!r}."
+            )
+            return to_folder
+
+        # do not call self._join_bucket_folder for list_blobs
+        # because it's built-in to that method already!
+        # however, we still need to do it because we're using relative_to
+        bucket_folder = self._join_bucket_folder(from_folder)
+        for blob in blobs:
+            bucket_path = Path(blob.name).relative_to(bucket_folder)
+            if bucket_path.is_dir():
+                continue
+            to_path = to_folder / bucket_path
+            to_path.parent.mkdir(parents=True, exist_ok=True)
+            self.logger.info(
+                f"Downloading blob from bucket {self.bucket!r} path {bucket_path!r}"
+                f"to {to_path}."
+            )
+            await run_sync_in_worker_thread(
+                blob.download_to_filename, filename=str(to_path), **download_kwargs
+            )
+        return to_folder
```

Comment:
Should these be absolute paths?

Reply:
I think that would be more explicit and less prone to error.